Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/CertificateUtil.java

```diff
@@ -76,4 +97,149 @@ private void closeStream(Closeable stream) {
             LOG.warn("Unable to close the stream {}", ex.getLocalizedMessage(), ex);
         }
     }
+
+    public static KeyStore loadKeyStore(final String storeType, final String password, final String storeLoc)
+        throws CertificateManagerException {
+        InputStream is = null;
+        KeyStore secretStore = null;
+        try {
+            secretStore = KeyStore.getInstance(storeType);
+            if (!PKCS11_TYPE.equalsIgnoreCase(storeType)) {
+                is = new FileInputStream(storeLoc);
+            }
+            secretStore.load(is, password.toCharArray());
+        } catch (final IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException ex) {
+            LOG.error(KEYSTORE_ERROR_MSG, ex.getLocalizedMessage(), ex);
+            throw new CertificateManagerException(ex.getMessage(), ex);
+        } finally {
+            try {
```

Comment:
You have a method immediately above this one named closeStream(). Either use that, or use a try-with-resources block.